controllers/deactivation/mapper.go

```diff
@@ -4,33 +4,32 @@ import (
 	"errors"
 
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-type UserSignupToMasterUserRecordMapper struct{}
-
-var _ handler.Mapper = UserSignupToMasterUserRecordMapper{}
 var mapperLog = ctrl.Log.WithName("UserSignupToMasterUserRecordMapper")
 
-func (b UserSignupToMasterUserRecordMapper) Map(obj handler.MapObject) []reconcile.Request {
-	if userSignup, ok := obj.Object.(*toolchainv1alpha1.UserSignup); ok {
+func MapUserSignupToMasterUserRecord() func(object client.Object) []reconcile.Request {
```

Comment:
right, but that cannot be applied to all cases - like this one: 
https://github.com/codeready-toolchain/host-operator/blob/f5ae3d9ab660036c8fc2968576ee501a5235cb2a/controllers/usersignup/mapper.go#L16-L17
or this one:
https://github.com/codeready-toolchain/toolchain-common/blob/79793648c8cbc9ed196035606ccc2cfeb583a55b/controllers/label_event_handler.go#L13-L14
and I thought that it would be better to have it more consistent

Reply:
ok, that's a valid point ;)